jt-warp/src/main/java/it/geosolutions/jaiext/warp/WarpOpImage.java

```diff
@@ -209,11 +214,21 @@ public Raster computeTile(final int tileX, final int tileY) {
         }
 
         // are we outside the roi
-        if (roi != null && !roi.intersects(srcRect)) {
-            if (setBackground) {
-                ImageUtil.fillBackground(dest, destRect, backgroundValues);
+        if (roi != null) {
+            boolean optmizedCheck = roi instanceof ROIGeometry || roi instanceof ROIShape;
+            if (!optmizedCheck) {
+                Rectangle rect = roi.getBounds();
+                // The roi.intersects call will be a getData() if the roi is a 
+                // pure ROI image. If the roi is big, that getData may be an overkill
+                // so let's skip the optimization when this happens
+                optmizedCheck = (rect.getWidth() * rect.getHeight() < THRESHOLD);
```

Comment:
Maybe reduce it to a bbox check instead?

Reply:
this happens right afterwards in computeRect.